eZ/Publish/Core/MVC/Symfony/View/Builder/ContentViewBuilder.php

```diff
@@ -130,7 +130,7 @@ public function buildView(array $parameters)
                 throw new InvalidArgumentException('Location', 'Provided location does not belong to selected content');
             }
 
-            if (isset($parameters['contentId']) && $location->contentId !== $parameters['contentId']) {
+            if (isset($parameters['contentId']) && $location->contentId != $parameters['contentId']) {
```

Comment:
Please cast it to `int` instead of doing weak comparison.

Reply:
fixed in 46790a2